### PR TITLE
👩🏾 Add sidebar contacts endpoint

### PR DESCRIPTION
**Pull request by pguduguntla**
## Status:
:rocket: Ready

## Description :star2:

Gets all the unique contacts based on when they were last accessed.
Data looks like this:

image

### CODE REVIEW

File: backend/api/views/messages.py [hunk 1]
```diff
@@ -138,6 +140,56 @@ def contact_mentor(mentor_id):
     return create_response(status=200, message="successfully sent email message")
 
 
+@messages.route("/contacts/<string:user_id>", methods=["GET"])
+def get_sidebar(user_id):
+    try:
+        sentMessages = DirectMessage.objects.filter(
+            Q(sender_id=user_id) | Q(recipient_id=user_id)
+        ).order_by("-created_at")
```
Comment (zfaizal2) [inline backend/api/views/messages.py:148]: fyi mongo orderers by "created at" already!

**Review by zfaizal2 — APPROVED** [review]
noice!

File: backend/api/views/messages.py [hunk 2]
```diff
@@ -138,6 +140,56 @@ def contact_mentor(mentor_id):
...
+
+        contacts = []
+        sidebarContacts = set()
+        for message in sentMessages:
+            otherId = message["recipient_id"]
```
Comment (Lam7150) [inline backend/api/views/messages.py:153]: what is otherId?

File: backend/api/views/messages.py [hunk 3]
```diff
@@ -138,6 +140,56 @@ def contact_mentor(mentor_id):
...
+                except:
+                    pass
+                if not otherUser:
+                    try:
+                        otherUser = MenteeProfile.objects.get(id=otherId)
+                    except:
+                        msg = "Could not find mentor or mentee for given ids"
+                        return create_response(status=422, message=msg)
+
+                otherUser = json.loads(otherUser.to_json())
```
Comment (Lam7150) [inline backend/api/views/messages.py:170]: Nit: Add comments here to make it more readable what is happening in this section of code

**Review by Lam7150 — APPROVED** [review]
Looks good to me! Comments are not blocking

**Pull request merged**